### DISCUSSION TITLE
Create updated instance of class that has lombok builder

### DISCUSSION THREAD

**Discussion by treecoder01** [mapstruct/mapstruct#3020] [Q&A]
Given a class ('User' below) that is immutable and has lombok builder defined. I want to create an updated instance that is the 'existing' updated with the 'incoming', similar to how its done with a mutable class:

![image](https://user-images.githubusercontent.com/34738467/189584845-35f71e0b-bd54-432e-8732-2db3291d8423.png)

Where the return value is the updated instance.

If this is not supported with mapstruct currently, has such functionality been proposed yet?

Since I have a large number of such classes that also has nested fields, also with lombok builders, I would love to make this work.

**Comment by sjaakd** [comment]
Have you tried passing the builder?

Op ma 12 sep. 2022 08:06 schreef Kristofer ***@***.***>:

**Reply by treecoder01** [reply]
Good idea. I tried it, but it seems to not work since the builders do not have getters for the properties required for update.

I also tried to have package private setters on my classes (to be mapped), but those are not found by mapstruct :/
